lib/Doctrine/DBAL/Types/JsonArrayType.php

```diff
@@ -70,12 +51,5 @@ public function getName()
     {
         return Type::JSON_ARRAY;
     }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function requiresSQLCommentHint(AbstractPlatform $platform)
```

Comment:
I'm not sure about the right one to keep uncommented. It should probably be the old one for BC reasons.

But the native type can only be mapped to a single doctrine type, and all other doctrine types reusing the same native type must use a comment hint to make the schema retrieval work properly.
Otherwise the native field will always be considered as a ``json_array`` and any of your ``json`` field will trigger an ``ALTER TABLE`` query in each schema update run or migration generation. this is precisely why we introduced comment hints